### ISSUE TITLE
SteamOS - Game Mode: Wi-Fi Auto-Connect toggle resets to On(Always)

### ISSUE THREAD

**Issue by YoreonX** [ValveSoftware/SteamOS#1284]
# System Prerequisites:

Device: Steam Deck OLED

OS Channel Update: Main

Steam Client Update Channel: Steam Deck Beta

OS Version: 3.6

Kernel Version: 6.1.52-valve9-1-neptune-61

BIOS Version: F7G0105

Steam Version: 1701289036

# Steps to Reproduce:

1. Be in SteamOS "Game Mode"

(This works regardless of SteamOS Version)

2. Press the Steam Button --> Go to Settings

3. Select Internet

4. Go to an existing saved WiFi Network or Add a new Network

5. Press "A" for more info

6. Under Disconnect, Toggle off "Autoconnect with this network"

7. Press "B" to return to the previous menu

8. Press "A" to reopen Network "Info" and it should be Re-toggled On.

https://github.com/ValveSoftware/SteamOS/assets/128438463/70ac6c3d-b5b6-4e34-b95b-c2323c0820fb

# Notes

Auto-Connect has never worked since SteamOS's launch, specifically in Game Mode.

This is fine largely, and this happens on all Steam Decks, all versions of SteamOS, so it should be easy to reproduce regardless of device.

I'm posting this here to bring awareness, because I don't think anyone has noticed this toggle has never worked since or atleast have never made a formal issue on the matter.

So hopefully this gets the ball rolling, thanks for the read & I hope this is useful information to help us come to a solution.

# EDIT:

_i added the video, in case there was any confusion._

**Comment by RedAsKetchum** [comment]
They dont care. Its been 20 days of the WIFI reports getting ignored.

**Comment by YoreonX** [comment]
> They dont care. Its been 20 days of the WIFI reports getting ignored. 

Hello, the issue you are experiencing is not the same as this thread.

I am the only person to notice this bug in all of Valve's Github experience.

Please only reply here if you are a developer or willing to reproduce the issue here so there can be more data on this matter for the developers to work with.

**Comment by HauieT** [comment]
Yup, happens to me too. I can disable autoconnect through desktop, but once you manually connect, the auto toggle is toggled back to autoconnect

**Comment by jtentmusic** [comment]
Having the same issue

DEVICE: STEAM DECK OLED 

https://github.com/ValveSoftware/SteamOS/assets/158021656/4d67b47c-1c7f-4117-9d87-1c7725692209

**Comment by KirbyQK** [comment]
I'm also having this issue on an OLED with OS 3.5.19

**Comment by MaelBriantin** [comment]
Same problem here. I can disable auto-connect in desktop mode, but not in game mode. Toggle the switch do nothing.

**Comment by Samael1408** [comment]
I noticed it too, I have 5 GHz and 2.4 GHz Wi-Fi and I can't turn off auto-connect for 5 GHz. I have to manually change to 2.4 GHz Wi-Fi when is 5 GHz signal is too weak. And steam deck prioritizes 5 GHz even if signal is weak.

**Comment by Vlaphor** [comment]
OLED model here, still happening as well.

**Comment by YoreonX** [comment]
https://github.com/user-attachments/assets/68286ba4-686e-45a8-b27c-e8bf7edc7afb

It appears this bug has been resolved, if I could just get someone else to confirm Autoconnect is also functioning properly I will close this thread as resolved.

Thank you for your time.

**Comment by Samael1408** [comment]
Indeed this bug has been fixed.

**Comment by YoreonX** [comment]
> Indeed this bug has been fixed.

Thank you for taking time out of your day to assist with additional testing.

As of..

SteamOS Build: 20250218.100

OS Version 3.8

Steam Version: 1738026274

This issue has been resolved, and thus I will close the issue.

I hope for the Stable launch of SteamOS 3.8 this fix is included in the patch notes.

Thank you everyone who replied & to the developers who took the time to resolve this issue.

And to Kisak-Valve's Moderation efforts.

I hope everyone has a great weekend!!
